uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/work-pattern-api.js

```diff
@@ -13,7 +13,43 @@ define([
     return api.extend({
 
       /**
-       * This method returns the calendar for the given contact(s) and period,
+       * Assigns a work pattern to a contact
+       *
+       * @param {string} contactId
+       * @param {string} workPatternID
+       * @param {string} effectiveDate
+       * @param {string} effectiveEndDate
+       * @param {string} changeReason
+       * @param {object} params - additional parameters
+       * @return {Promise} Resolved with {Array} All Work Patterns
+       */
+      assignWorkPattern: function (contactId, workPatternID, effectiveDate, effectiveEndDate, changeReason, params) {
```

Comment:
@AkA84 @davialexandre Yes wrong example sorry, here is the correct one https://github.com/civicrm/civihr/blob/57a3415ee63faa8632c31a1fd2028a389679d54c/uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/leave-request-api.js#L153

I think that specifying the unique parameters for this API explicitly is better because then its easier to just refer the API(JS Code) documentation to see which parameters I need to send to assign a work pattern, otherwise we would need to go to API explorer everytime and find out what the required parameters are, also we need to get their names.